### PR TITLE
Refactored generics in web layer.

### PR DESCRIPTION
**Pull request by ivanCosic**
Removed wildcard types, and redundant type declarations inside generics.

### CODE REVIEW

File: web/src/main/java/eu/execom/labs/pomodorospringreact/controller/PomodoroController.java [hunk 1]
```diff
@@ -21,34 +21,34 @@
...
+        return new ResponseEntity<>(pomodoroService.find(pomodoroId), HttpStatus.OK);
     }
 
     @RequestMapping(method = RequestMethod.POST, value = "/user/{id}/start")
     public ResponseEntity<PomodoroDto> startPomodoro(@PathVariable("id") long userId,
                                                      @RequestParam(value = "duration", required = false) Integer duration,
                                                      @RequestParam(value = "shortPauseDuration", required = false) Integer shortPauseDuration,
                                                      @RequestParam(value = "longPauseDuration", required = false) Integer longPauseDuration) {
-        return new ResponseEntity<PomodoroDto>(pomodoroService.start(userId, duration, shortPauseDuration, longPauseDuration), HttpStatus.OK);
+        return new ResponseEntity<>(pomodoroService.start(userId, duration, shortPauseDuration, longPauseDuration), HttpStatus.OK);
```
Comment (dejanjesic) [inline web/src/main/java/eu/execom/labs/pomodorospringreact/controller/PomodoroController.java]: Formatting

File: web/src/main/java/eu/execom/labs/pomodorospringreact/util/GlobalControllerExceptionHandler.java [hunk 2]
```diff
@@ -26,7 +26,7 @@
      */
     @ExceptionHandler({EntityNotFoundException.class})
     @ResponseBody
-    public ResponseEntity<?> exceptionHandler(HttpServletRequest request, EntityNotFoundException exception) {
+    public ResponseEntity<Map<String, String>> exceptionHandler(HttpServletRequest request, EntityNotFoundException exception) {
```
Comment (dejanjesic) [inline web/src/main/java/eu/execom/labs/pomodorospringreact/util/GlobalControllerExceptionHandler.java]: Formatting

File: web/src/main/java/eu/execom/labs/pomodorospringreact/util/GlobalControllerExceptionHandler.java [hunk 3]
```diff
@@ -37,7 +37,7 @@
      */
     @ExceptionHandler({InvalidTokenException.class})
     @ResponseBody
-    public ResponseEntity<?> exceptionHandler(final HttpServletRequest request, final InvalidTokenException exception) {
+    public ResponseEntity<Map<String, String>> exceptionHandler(final HttpServletRequest request, final InvalidTokenException exception) {
```
Comment (dejanjesic) [inline web/src/main/java/eu/execom/labs/pomodorospringreact/util/GlobalControllerExceptionHandler.java]: Formatting

File: web/src/main/java/eu/execom/labs/pomodorospringreact/util/GlobalControllerExceptionHandler.java [hunk 4]
```diff
@@ -48,11 +48,9 @@
      * @param exception Exception
      * @return ResponseEntity
      */
-    @ExceptionHandler({
-            ServiceException.class,
-    })
+    @ExceptionHandler({ServiceException.class})
     @ResponseBody
-    public ResponseEntity<?> exceptionHandler(HttpServletRequest request, ServiceException exception) {
+    public ResponseEntity<Map<String, String>> exceptionHandler(HttpServletRequest request, ServiceException exception) {
```
Comment (dejanjesic) [inline web/src/main/java/eu/execom/labs/pomodorospringreact/util/GlobalControllerExceptionHandler.java]: Formatting

File: web/src/main/java/eu/execom/labs/pomodorospringreact/util/GlobalControllerExceptionHandler.java [hunk 5]
```diff
@@ -78,12 +76,12 @@
      * @param httpStatus Http status
      * @return ResponseEntity
      */
-    private ResponseEntity<?> createResponseEntity(HttpServletRequest request, Exception exception, HttpStatus httpStatus) {
+    private ResponseEntity<Map<String, String>> createResponseEntity(HttpServletRequest request, Exception exception, HttpStatus httpStatus) {
```
Comment (dejanjesic) [inline web/src/main/java/eu/execom/labs/pomodorospringreact/util/GlobalControllerExceptionHandler.java]: Formatting

File: web/src/main/java/eu/execom/labs/pomodorospringreact/controller/PomodoroController.java (hunk 1 above)
Comment (b-laci) [reply web/src/main/java/eu/execom/labs/pomodorospringreact/controller/PomodoroController.java]: Also change HttpStatus -> created

File: web/src/main/java/eu/execom/labs/pomodorospringreact/controller/TaskController.java [hunk 6]
```diff
@@ -31,9 +31,8 @@
     }
 
     @RequestMapping(method = RequestMethod.PUT, value = "/{id}")
-    public ResponseEntity<TaskDto> updateTask(
-            @PathVariable("id") long taskId,
-            @RequestBody UpdateTaskDto updateTaskDto) {
+    public ResponseEntity<TaskDto> updateTask(@PathVariable("id") long taskId,
```
Comment (dejanjesic) [inline web/src/main/java/eu/execom/labs/pomodorospringreact/controller/TaskController.java]: We should use "resourceId" all the time

File: web/src/main/java/eu/execom/labs/pomodorospringreact/controller/TeamController.java [hunk 7]
```diff
@@ -54,12 +54,14 @@
     }
 
     @RequestMapping(method = RequestMethod.DELETE, value = "/{teamId}/remove-user/{userId}")
-    public ResponseEntity<Boolean> leaveTeam(@PathVariable("teamId") long teamId, @PathVariable("userId") long userId) {
+    public ResponseEntity<Boolean> leaveTeam(@PathVariable("teamId") long teamId,
```
Comment (dejanjesic) [inline web/src/main/java/eu/execom/labs/pomodorospringreact/controller/TeamController.java]: HttpStatus should be 'DELETED'

**Pull request merged**